src/Mod/Sketcher/Gui/DrawSketchHandler.h

```diff
@@ -225,6 +225,15 @@ class SketcherGuiExport DrawSketchHandler
     virtual void onWidgetChanged()
     {}
 
+    virtual bool shouldDrawPositionAtCursor() const

```

Comment:
And here we are at the source.

Here the function offers an option to handlers to override the logic to draw or not to draw.

The functionality to draw is also offered by this class some lines below. It draws or not according to a preferences parameter.

The caller to draw is, ultimately the handler (e.g. DSHLine).

So DSHLine asks grandpa to draw (according to preferences), and uses a facility provided by granpa to override the logic for drawing... a logic that is available at the handler (via the controller which is a member of the handler).

A first approach that I think it is perhaps more straightforward, is to make the drawing functions below virtual, thereby allowing any handler to overload behaviour.